### PR TITLE
Integrate renderer

### PR DESCRIPTION
**Pull request by omni-viral**
* Rebase _"renderer"_ branch onto _"develop"_
* `MeshComponent`, `MaterialComponent`, `LightComponent`
* `MechBuilder`, `TextureBuilder` and `MaterialBuilder` can be `Send + Sync + 'static` now
* Finishing `MechBuilder` and `MaterialBuilder` in `RenderSystem`
* Populating `Scene` on each frame. Slow. Need rework here.
* Examples fix
  - [x] 00_hello_world
  - [x] 01_window
  - [x] 02_sphere
  - [ ] 03_renderable
  - [x] 04_pong
  - [ ] 05_assets

### CODE REVIEW

File: amethyst_renderer/src/config.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //! Renderer configuration.
 
-use winit::WindowBuilder;
+use glutin::WindowBuilder;
 
 /// Structure for holding the renderer configuration.
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
```
Comment (torkleyy) [inline amethyst_renderer/src/config.rs]: I'm pretty sure we need defaults for this, but it's not a priority.

**Comment by torkleyy** [comment]
Yay, thank you! Let's merge this ;)

**Comment by msiglreith** [comment]
What's the motivation for `winit`->`glutin`?

**Comment by omni-viral** [comment]
> What's the motivation for winit->glutin?
No reason.
Actually it's just merging artifacts.
Should be `winit` where possible.

**Pull request merged**